--- /home/user/simics-qsp-x86-6.0.65/targets/qsp-x86/qsp-uefi.include	2022-12-08 17:50:32.000000000 -0500
+++ /home/user/simics-qsp-x86-6.0.65/targets/qsp-x86/modified.include	2023-03-19 12:45:36.790462733 -0400
@@ -1,6 +1,6 @@
 decl {
  params from "qsp-images.include"
   default bios_image =
-   "%simics%/targets/qsp-x86/images/SIMICSX58IA32X64_1_1_0_r.fd"
+   "%simics%/targets/qsp-x86/images/BOARDX58ICH10.fd"
   default enable_efi = TRUE
 }
--- /home/user/simics-qsp-cpu-6.0.12/targets/qsp-x86/original_qsp-modern.simics	2023-03-28 23:28:53.148343074 -0400
+++ /home/user/simics-qsp-cpu-6.0.12/targets/qsp-x86/qsp-modern-core.simics	2023-03-28 13:13:41.187540505 -0400
@@ -9,5 +9,6 @@
   default num_cores = 2
   default num_threads = 2
 }
+$disk1_image="%simics%/targets/qsp-x86/images/TestVHD.vhd"
 
 run-command-file "%simics%/targets/qsp-x86/qsp-clear-linux.simics"
