--- test/edk2/MdeModulePkg/MdeModulePkg.dec	2023-03-28 22:57:01.828366669 -0400
+++ src/edk2/MdeModulePkg/MdeModulePkg.dec	2023-03-28 11:52:09.769492749 -0400
@@ -9,7 +9,6 @@
 # (C) Copyright 2016 - 2019 Hewlett Packard Enterprise Development LP<BR>
 # Copyright (c) 2017, AMD Incorporated. All rights reserved.<BR>
 # Copyright (c) Microsoft Corporation.<BR>
-# Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.<BR>
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 #
 ##
@@ -118,10 +117,6 @@
   #
   IpmiLib|Include/Library/IpmiLib.h
 
-  ## @libraryclass  Provides interfaces to send/receive IPMI command.
-  #
-  IpmiCommandLib|Include/Library/IpmiCommandLib.h
-
   ## @libraryclass  Provides interfaces for platform to return root bridge information to PciHostBridgeDxe driver.
   #
   PciHostBridgeLib|Include/Library/PciHostBridgeLib.h
@@ -164,6 +159,9 @@
   VariableFlashInfoLib|Include/Library/VariableFlashInfoLib.h
 
 [Guids]
+  gEfiSmmHardenBootServiceGuid        = { 0xccee1234, 0xce70, 0x4bb0, { 0x98, 0x5d, 0x5e, 0x7b, 0x14, 0x0d, 0xe4, 0xd2 }}
+  gEfiSmmHardenVariableManagerGuid    = { 0x82fbc314, 0xf00d, 0xacd0, { 0xf8, 0x89, 0x45, 0x81, 0x17, 0x2d, 0xa4, 0xd3 }}
+  gEfiSmmHardenExampleVariableGuid    = { 0xf7327a91, 0xE4D2, 0x4A93, { 0xB2, 0x42, 0x51, 0xf8, 0x3F, 0x02, 0x58, 0x4B }}
   ## MdeModule package token space guid
   # Include/Guid/MdeModulePkgTokenSpace.h
   gEfiMdeModulePkgTokenSpaceGuid       = { 0xA1AFF049, 0xFDEB, 0x442a, { 0xB3, 0x20, 0x13, 0xAB, 0x4C, 0xB7, 0x2B, 0xBC }}
@@ -442,9 +440,6 @@
   gBootDiscoveryPolicyMgrFormsetGuid = { 0x5b6f7107, 0xbb3c, 0x4660, { 0x92, 0xcd, 0x54, 0x26, 0x90, 0x28, 0x0b, 0xbd } }
 
 [Ppis]
-  ## Include/Ppi/FirmwareVolumeShadowPpi.h
-  gEdkiiPeiFirmwareVolumeShadowPpiGuid = { 0x7dfe756c, 0xed8d, 0x4d77, {0x9e, 0xc4, 0x39, 0x9a, 0x8a, 0x81, 0x51, 0x16 } }
-
   ## Include/Ppi/AtaController.h
   gPeiAtaControllerPpiGuid       = { 0xa45e60d1, 0xc719, 0x44aa, { 0xb0, 0x7a, 0xaa, 0x77, 0x7f, 0x85, 0x90, 0x6d }}
 
@@ -525,6 +520,7 @@
   gEdkiiPeiBootInCapsuleOnDiskModePpiGuid   = { 0xb08a11e4, 0xe2b7, 0x4b75, { 0xb5, 0x15, 0xaf, 0x61, 0x6, 0x68, 0xbf, 0xd1  } }
 
 [Protocols]
+  gExample1_Driver_LockboxProtocolGuid           = { 0xf3aae06f, 0x7716, 0x4722,{ 0xb6, 0x26, 0x02, 0x64, 0x79, 0xb9, 0xfa, 0x57 } }
   ## Load File protocol provides capability to load and unload EFI image into memory and execute it.
   #  Include/Protocol/LoadPe32Image.h
   #  This protocol is deprecated. Native EDKII module should NOT use this protocol to load/unload image.
@@ -925,13 +921,15 @@
   # @Prompt Enable process non-reset capsule image at runtime.
   gEfiMdeModulePkgTokenSpaceGuid.PcdSupportProcessCapsuleAtRuntime|FALSE|BOOLEAN|0x00010079
 
-[PcdsFeatureFlag.IA32, PcdsFeatureFlag.ARM, PcdsFeatureFlag.AARCH64, PcdsFeatureFlag.LOONGARCH64]
+[PcdsFeatureFlag.IA32, PcdsFeatureFlag.ARM, PcdsFeatureFlag.AARCH64]
   gEfiMdeModulePkgTokenSpaceGuid.PcdPciDegradeResourceForOptionRom|FALSE|BOOLEAN|0x0001003a
 
 [PcdsFeatureFlag.IA32, PcdsFeatureFlag.X64]
   ## Indicates if DxeIpl should switch to long mode to enter DXE phase.
+  #  It is assumed that 64-bit DxeCore is built in firmware if it is true; otherwise 32-bit DxeCore
+  #  is built in firmware.<BR><BR>
   #   TRUE  - DxeIpl will load a 64-bit DxeCore and switch to long mode to hand over to DxeCore.<BR>
-  #   FALSE - DxeIpl will load a 32-bit or 64-bit DxeCore and perform stack switch to hand over to DxeCore.<BR>
+  #   FALSE - DxeIpl will load a 32-bit DxeCore and perform stack switch to hand over to DxeCore.<BR>
   # @Prompt DxeIpl switch to long mode.
   gEfiMdeModulePkgTokenSpaceGuid.PcdDxeIplSwitchToLongMode|TRUE|BOOLEAN|0x0001003b
 
--- test/edk2/MdePkg/MdePkg.dec	2023-03-28 22:57:01.996366347 -0400
+++ src/edk2/MdePkg/MdePkg.dec	2023-03-28 11:52:57.257339449 -0400
@@ -7,9 +7,6 @@
 # Copyright (c) 2007 - 2022, Intel Corporation. All rights reserved.<BR>
 # Portions copyright (c) 2008 - 2009, Apple Inc. All rights reserved.<BR>
 # (C) Copyright 2016 - 2021 Hewlett Packard Enterprise Development LP<BR>
-# Copyright (c) 2022, Loongson Technology Corporation Limited. All rights reserved.<BR>
-# Copyright (c) 2021 - 2022, Arm Limited. All rights reserved.<BR>
-# Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.<BR>
 #
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 #
@@ -46,9 +43,6 @@
 [Includes.RISCV64]
   Include/RiscV64
 
-[Includes.LOONGARCH64]
-  Include/LoongArch64
-
 [LibraryClasses]
   ##  @libraryclass  Provides most usb APIs to support the Hid requests defined in Usb Hid 1.1 spec
   #                  and the standard requests defined in Usb 1.1 spec.
@@ -281,10 +275,6 @@
   ## @libraryclass  Provides function for SMM CPU Rendezvous Library.
   SmmCpuRendezvousLib|Include/Library/SmmCpuRendezvousLib.h
 
-  ##  @libraryclass  Provides services to generate Entropy using a TRNG.
-  #
-  ArmTrngLib|Include/Library/ArmTrngLib.h
-
 [LibraryClasses.IA32, LibraryClasses.X64, LibraryClasses.AARCH64]
   ##  @libraryclass  Provides services to generate random number.
   #
@@ -317,11 +307,8 @@
   ##  @libraryclass  Provides function to support TDX processing.
   TdxLib|Include/Library/TdxLib.h
 
-[LibraryClasses.RISCV64]
-  ##  @libraryclass  Provides function to make ecalls to SBI
-  BaseRiscVSbiLib|Include/Library/BaseRiscVSbiLib.h
-
 [Guids]
+  gEfiSmmHardenProtocolGuid   = { 0xccee1234, 0xce70, 0x4bb0, { 0x98, 0x5d, 0x5e, 0x7b, 0x14, 0x0d, 0xe4, 0xd2 }}
   #
   # GUID defined in UEFI2.1/UEFI2.0/EFI1.1
   #
@@ -419,13 +406,10 @@
   gEfiEventReadyToBootGuid       = { 0x7CE88FB3, 0x4BD7, 0x4679, { 0x87, 0xA8, 0xA8, 0xD8, 0xDE, 0xE5, 0x0D, 0x2B }}
 
   ## Include/Guid/EventGroup.h
-  gEfiEventAfterReadyToBootGuid  = { 0x3a2a00ad, 0x98b9, 0x4cdf, { 0xa4, 0x78, 0x70, 0x27, 0x77, 0xf1, 0xc1, 0x0b }}
-
-  ## Include/Guid/EventGroup.h
   gEfiEventMemoryMapChangeGuid   = { 0x78BEE926, 0x692F, 0x48FD, { 0x9E, 0xDB, 0x01, 0x42, 0x2E, 0xF0, 0xD7, 0xAB }}
 
   ## Include/Guid/EventGroup.h
-  gEfiEventVirtualAddressChangeGuid   = { 0x13FA7698, 0xC831, 0x49C7, { 0x87, 0xEA, 0x8F, 0x43, 0xFC, 0xC2, 0x51, 0x96 }}
+  gEfiEventVirtualAddressChangeGuid = { 0x13FA7698, 0xC831, 0x49C7, { 0x87, 0xEA, 0x8F, 0x43, 0xFC, 0xC2, 0x51, 0x96 }}
 
   ## Include/Guid/EventGroup.h
   gEfiEventBeforeExitBootServicesGuid = { 0x8BE0E274, 0x3970, 0x4B44, { 0x80, 0xC5, 0x1A, 0xB9, 0x50, 0x2F, 0x3B, 0xFC }}
@@ -1923,9 +1907,6 @@
   ## Include/Protocol/RedfishDiscover.h
   gEfiRedfishDiscoverProtocolGuid      = { 0x5db12509, 0x4550, 0x4347, { 0x96, 0xb3, 0x73, 0xc0, 0xff, 0x6e, 0x86, 0x9f }}
 
-  ## Include/Protocol/MemoryAttribute.h
-  gEfiMemoryAttributeProtocolGuid = { 0xf4560cf6, 0x40ec, 0x4b4a, { 0xa1, 0x92, 0xbf, 0x1d, 0x57, 0xd0, 0xb1, 0x89 }}
-
   #
   # Protocols defined in Shell2.0
   #
@@ -2343,11 +2324,6 @@
   # @Prompt Memory Address of GuidedExtractHandler Table.
   gEfiMdePkgTokenSpaceGuid.PcdGuidedExtractHandlerTableAddress|0x1000000|UINT64|0x30001015
 
-  ## This value is the IPMI KCS Interface I/O base address used to transmit IPMI commands.
-  #  The value of 0xca2 is the default I/O base address defined in IPMI specification.
-  # @Prompt IPMI KCS Interface I/O Base Address
-  gEfiMdePkgTokenSpaceGuid.PcdIpmiKcsIoBaseAddress|0xca2|UINT16|0x00000031
-
 [PcdsFixedAtBuild, PcdsPatchableInModule, PcdsDynamic, PcdsDynamicEx]
   ## This value is used to set the base address of PCI express hierarchy.
   # @Prompt PCI Express Base Address.
--- test/edk2-platforms/Platform/Intel/SimicsOpenBoardPkg/BoardX58Ich10/OpenBoardPkg.fdf	2023-03-28 22:57:59.736254814 -0400
+++ src/edk2-platforms/Platform/Intel/SimicsOpenBoardPkg/BoardX58Ich10/OpenBoardPkg.fdf	2023-03-28 21:25:51.113365510 -0400
@@ -41,6 +41,7 @@
 #NV_FTW_SPARE
 
 0x00080000|0x0016C000
+#0x00080000|0x00200000
 FV = FVMAIN_COMPACT
 
 $(SECFV_OFFSET)|$(SECFV_SIZE)
@@ -67,9 +68,11 @@
 NumBlocks     = $(CODE_BLOCKS)
 
 0x00000000|0x0016C000
+#0x00000000|0x00200000
 FV = FVMAIN_COMPACT
 
 0x0016C000|$(SECFV_SIZE)
+#0x00200000|$(SECFV_SIZE)
 FV = FvTempMemorySilicon
 
 [FD.MEMFD]
@@ -202,11 +205,15 @@
 INF  MdeModulePkg/Universal/Acpi/S3SaveStateDxe/S3SaveStateDxe.inf
 INF  MdeModulePkg/Universal/Acpi/BootScriptExecutorDxe/BootScriptExecutorDxe.inf
 INF  UefiCpuPkg/CpuS3DataDxe/CpuS3DataDxe.inf
-INF  MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBox.inf
+#INF  MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBox.inf
 INF  MdeModulePkg/Universal/Console/TerminalDxe/TerminalDxe.inf
 INF  MdeModulePkg/Universal/PrintDxe/PrintDxe.inf
 INF  MdeModulePkg/Bus/Scsi/ScsiBusDxe/ScsiBusDxe.inf
 INF  MdeModulePkg/Bus/Scsi/ScsiDiskDxe/ScsiDiskDxe.inf
+INF  MdeModulePkg/Application/BBClient/BBClient.inf
+INF  MdeModulePkg/Application/Example1_Driver_Lockbox/Example1_Driver_Lockbox.inf
+INF  MdeModulePkg/Application/Example1_App/Example1_App.inf
+INF  MdeModulePkg/Application/SmmHarden/SmmHarden.inf
 
 INF  MdeModulePkg/Bus/Ata/AtaAtapiPassThru/AtaAtapiPassThru.inf
 INF  MdeModulePkg/Bus/Ata/AtaBusDxe/AtaBusDxe.inf
--- test/edk2-platforms/Platform/Intel/SimicsOpenBoardPkg/BoardX58Ich10/OpenBoardPkg.dsc	2023-03-28 22:57:59.736254814 -0400
+++ src/edk2-platforms/Platform/Intel/SimicsOpenBoardPkg/BoardX58Ich10/OpenBoardPkg.dsc	2023-03-28 12:10:05.339141872 -0400
@@ -251,11 +251,16 @@
   MdeModulePkg/Universal/Console/TerminalDxe/TerminalDxe.inf
   MdeModulePkg/Universal/EbcDxe/EbcDxe.inf
   MdeModulePkg/Universal/PrintDxe/PrintDxe.inf
+  MdeModulePkg/Application/BBClient/BBClient.inf
+  MdeModulePkg/Application/Example1_Driver_Lockbox/Example1_Driver_Lockbox.inf 
+  MdeModulePkg/Application/Example1_App/Example1_App.inf
+  MdeModulePkg/Application/SmmHarden/SmmHarden.inf
+  
 !if gMinPlatformPkgTokenSpaceGuid.PcdBootToShellOnly == FALSE
-  MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBox.inf {
-    <LibraryClasses>
-      LockBoxLib|MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxSmmLib.inf
-  }
+  #MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBox.inf {
+  #  <LibraryClasses>
+  #    LockBoxLib|MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxSmmLib.inf
+  #}
   UefiCpuPkg/CpuS3DataDxe/CpuS3DataDxe.inf
   UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.inf
 !endif
